### ISSUE TITLE
Python 3.8: Error in import

### ISSUE THREAD

**Issue by joshkel** [CoolProp/CoolProp#2078]
### Description

I'm trying to install CoolProp on an M1 MacBook, running macOS 12.1 and macOS's Python 3.8.9. There's no pre-built version available, so `pip install coolprop` builds from source. 

My initial install attempt failed with the following error:

```
    error: $MACOSX_DEPLOYMENT_TARGET mismatch: now "10.9" but "10.14" during configure
```

This appears to be the same issue that's reported in #2003 (and so I'm not worried about it for the purposes of this issue).

I retried the installation, explicitly setting MACOS_DEPLOYMENT_TARGET. This built and installed, but when I go to import the package, I get the following error:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/josh/tmp/venv/lib/python3.8/site-packages/CoolProp/__init__.py", line 22, in <module>
    __fluids__ = CoolProp.get_global_param_string('fluids_list').split(',')
  File "CoolProp.pyx", line 307, in CoolProp.CoolProp.get_global_param_string
  File "stringsource", line 15, in string.from_py.__pyx_convert_string_from_py_std__in_string
TypeError: expected bytes, str found
```

Which appears to be the same issue as #1876 - although that issue says that the problem is fixed in CoolProp 6.4.0, and I'm using 6.4.1.

### Steps to Reproduce

```sh
pip3 cache purge
rm -rf venv
python3 -m venv venv
. venv/bin/activate
MACOSX_DEPLOYMENT_TARGET=12 pip install coolprop
python -m CoolProp
```

**Expected behavior:** Successful installation and import of CoolProp. (The `python -m CoolProp` command will fail with an expected error, "'CoolProp' is a package and cannot be directly executed.")

**Actual behavior:** "TypeError: expected bytes, str found." See above for a more complete call stack.

### Versions

**CoolProp Version:** 6.4.1
**Operating System and Version:** macOS 12.1
**Access Method:** Python 3.8.9

### Additional information

Installing from source appears to work.

**Comment by ibell** [comment]
Once installed,  move to another directory before testing, and don't pass
the -m flag

**Comment by joshkel** [comment]
@ibell I'm afraid I don't understand. I'm using pip to install to a Python virtual environment; the directory that I'm in doesn't matter. (At any rate, I tried changing to another directory, and it didn't help.)

I apologize if I didn't explain well enough; I used `python -m CoolProp` to have a one-line way of demonstrating the import error from a shell command. I can reproduce the same error from Python's REPL:

```
(venv) ➜  ~ python
Python 3.8.9 (default, Oct 26 2021, 07:25:53)
[Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import CoolProp
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/josh/tmp/venv/lib/python3.8/site-packages/CoolProp/__init__.py", line 22, in <module>
    __fluids__ = CoolProp.get_global_param_string('fluids_list').split(',')
  File "CoolProp.pyx", line 307, in CoolProp.CoolProp.get_global_param_string
  File "stringsource", line 15, in string.from_py.__pyx_convert_string_from_py_std__in_string
TypeError: expected bytes, str found
```

If I run `python CoolProp` without the `-m` flag, I get the following error, which I'd expect:

```
/Users/josh/tmp/venv/bin/python: can't open file 'CoolProp': [Errno 2] No such file or directory
```

**Comment by ibell** [comment]
Hmm how did you acquire coolprop and precisely which version did you get? I
think this is somehow related to virtual environment and or pip because I
haven't seen similar reports.

**Comment by joshkel** [comment]
I acquired CoolProp by running `pip install`. I got 6.4.1. Here's the full output.

```
(venv) ➜  tmp MACOSX_DEPLOYMENT_TARGET=12 pip install coolprop
Collecting coolprop
  Downloading CoolProp-6.4.1.tar.gz (12.9 MB)
     |████████████████████████████████| 12.9 MB 2.6 MB/s
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: coolprop
  Building wheel for coolprop (setup.py) ... done
  Created wheel for coolprop: filename=CoolProp-6.4.1-cp38-cp38-macosx_12_0_arm64.whl size=4186022 sha256=d839583555cddf00ebec4b228198479e0802fad5d0d0dda93e5142ce37173237
  Stored in directory: /Users/josh/Library/Caches/pip/wheels/df/50/70/d903e2f53ce44c5580946559eec3436282ebb5242b69cc5c56
Successfully built coolprop
Installing collected packages: coolprop
Successfully installed coolprop-6.4.1
(venv) ➜  tmp python
Python 3.8.9 (default, Oct 26 2021, 07:25:53)
[Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import CoolProp
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/josh/tmp/venv/lib/python3.8/site-packages/CoolProp/__init__.py", line 22, in <module>
    __fluids__ = CoolProp.get_global_param_string('fluids_list').split(',')
  File "CoolProp.pyx", line 307, in CoolProp.CoolProp.get_global_param_string
  File "stringsource", line 15, in string.from_py.__pyx_convert_string_from_py_std__in_string
TypeError: expected bytes, str found
```

Using a virtual environment shouldn't make any difference.

I did some more testing, and it appears to be an issue with installing from source versus wheel. If I switch to an Intel Mac and use [SYSTEM_VERSION_COMPAT](https://stackoverflow.com/a/65402241/25507) so that pip is willing to use [one of the CoolProp wheels](https://pypi.org/project/CoolProp/#files):

```
SYSTEM_VERSION_COMPAT=1 pip install --only-binary :all: CoolProp
```

then it works. Conversely, if I disable binaries, I get the same `expected bytes, str found` error. (I observed this by running  `SYSTEM_VERSION_COMPAT=1 MACOSX_DEPLOYMENT_TARGET=10.14 pip install --no-binary :all: CoolProp` on an Intel Mac and `pip install --no-binary :all: CoolProp` on a Windows PC.) If no one else has reported, it's possible that they're on platforms where wheels are available.

**Comment by ibell** [comment]
What if you build directly from the development source? You can do something like this:
```
python -m pip install git+https://github.com/coolprop/coolprop.git
```
to checkout and build in one shot

I seem to recall there was another round of trying to fix these string issues

**Comment by joshkel** [comment]
The pip command you gave gives another error:

```
Collecting git+https://github.com/coolprop/coolprop.git
  Cloning https://github.com/coolprop/coolprop.git to /private/var/folders/j5/l20bwyp12yb180ql522y2t3w0000gn/T/pip-req-build-ldyyl3ln
  Running command git clone --filter=blob:none -q https://github.com/coolprop/coolprop.git /private/var/folders/j5/l20bwyp12yb180ql522y2t3w0000gn/T/pip-req-build-ldyyl3ln
  Resolved https://github.com/coolprop/coolprop.git to commit 339a7ac4977bace36947ef6608b68bc258afdc2c
  Running command git submodule update --init --recursive -q
ERROR: git+https://github.com/coolprop/coolprop.git does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.
```

If I locally clone the CoolProp repo, check out the v6.4.1 tag, and follow the [manual installation instructions](http://www.coolprop.org/coolprop/wrappers/Python/index.html#id3), it works.

I think the problem may be caused by the Cython version. The CoolProp .tar.gz sdist that was uploaded to PyPI has autoegnerated C++ source files that indicate that they were generated by Cython 0.29.2. When I clone and install CoolProp from source, I'm using Cython 0.29.26. #1876 links to https://github.com/cython/cython/issues/2819, which indicates that Cython had an issue with string conversions that was fixed in 0.29.6.

**Comment by ibell** [comment]
Of course the git solution with pip won't work (my bad), but I think you
find the solution,  build from source with updated cython.  We need to make
a new release...
